### PR TITLE
chore(database): add cbor for value encoding to reduce overall storage footprint

### PR DESCRIPTION
**Pull request by fuxingloh**
#### What kind of PR is this?:

/kind chore

#### What this PR does / why we need it:

http://cbor.io/ - a drop-in replacement for JSON

Reducing the overall storage footprint reduces the overall size of whale database snapshots allowing whale-api to startup faster. About 25% reduction based on the written test.

Related to https://github.com/DeFiCh/jellyfish/issues/979

### CODE REVIEW

**Comment by fuxingloh** [comment]
Problem with null & undefined handling.

> https://github.com/kawanet/msgpack-lite

> Note that both null and undefined are mapped to nil 0xC1 type. This means undefined value will be upgraded to null in other words.

**Review by chee-chyuan — APPROVED** [review]

File: src/module.database/provider.level/cbor.encoding.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+import cbor from 'cbor'
 
 /**
  * Faster & Smaller Value Codec for level.database.ts
  *
  * @see https://msgpack.org/
  * @see https://github.com/kawanet/msgpack-lite
  */
-export const MsgpackEncoding: CodecEncoder = {
+export const CborEncoding: CodecEncoder = {
```
Comment (canonbrother) [inline src/module.database/provider.level/cbor.encoding.ts:11]: we need update the desc above as well though

Comment (fuxingloh) [reply src/module.database/provider.level/cbor.encoding.ts:11]: It's updated with recent PR

Comment (fuxingloh) [reply src/module.database/provider.level/cbor.encoding.ts:11]: https://github.com/DeFiCh/whale/pull/732/commits/f8f3553e6b943735ec4b37b86767906104b4296b

**Review by canonbrother — APPROVED** [review]

**Review by monstrobishi — APPROVED** [review]
test looks good

**Pull request merged**